### ISSUE TITLE
Convert solution from DM format

### ISSUE THREAD

**Issue by Imran686** [casadi/casadi#3551]
After solving problem using IPOPT in CasADi, I am getting a solution vector in the format of DM as shown in the image attached. However, I wanted to convert the solution from the DM format to a normal format because its creating problems and cannot be saved to a torch.tensor. Please suggest. 
<img width="1436" alt="DM_opt" src="https://github.com/casadi/casadi/assets/72456599/b725b3cf-8cc5-4534-ab8c-ffa9d10f148f">

**Comment by jgillis** [comment]
When `m` is your DM matrix, do `np.array(m)` or `m.full()`

**Comment by Imran686** [comment]
Thanks for being so quick in responding. Yes I converted the DM matrix to numpy array and it worked as shown in the image attached. However, when I tried to convert it from numpy to torch.tensor it said:

**"TypeError: can't convert np.ndarray of type numpy.object_. The only supported types are: float64, float32, float16, complex64, complex128, int64, int32, int16, int8, uint8, and bool."**

The error is also shown in the image attached. Seems like the problem is that the data type of DM vector is object which is not changed even after converting it to the numpy matrix. 

Regarding using m.full() on a DM matrix m, it shows the similar error saying the .full() command cannot be used for the type object.  

<img width="1115" alt="DM_opt" src="https://github.com/casadi/casadi/assets/72456599/5abf4960-d4c1-4168-ba3f-e253d4ac585f">

**Comment by jgillis** [comment]
How did this structure come about to start with? Perhaps you can modify the code that generates it.

**Comment by Imran686** [comment]
Yes you are right. I have shown below my code. I have declared my solver as a structure. After that, I solved the optimization problem in parallel computation mode with different initializations and different constraints such that the solution for each initialization and constraint correspond to a separate processor. After that, I extracted the barrier coefficient (mu) using solver.stats() and concatenated it with the solution. So, this overall formulation might be the reason for such a structure?

<img width="429" alt="Code" src="https://github.com/casadi/casadi/assets/72456599/4e56ee60-c68f-4e86-8bb7-e92ef7919294">



I have also attached an image showing how the solution looks like after running this program.  

<img width="1442" alt="solved" src="https://github.com/casadi/casadi/assets/72456599/8a9c3331-98e4-49a7-bea1-b8efcb1ece23">
